integration/objectset_test.go

```diff
@@ -25,385 +24,450 @@ import (
+			objectSetRev2: defaultObjectSetRev2,
 		},
+		{
+			name: "with phase class",
+			objectSetRev1: func(cm1, cm2 *corev1.ConfigMap) *corev1alpha1.ObjectSet {
+				objectSet := defaultObjectSetRev1(cm1, cm2)
+				objectSet.Spec.Phases[0].Class = "default"
+				objectSet.Spec.Phases[1].Class = "default"
+				return objectSet
+			},
+			objectSetRev2: func(cm1, cm3 *corev1.ConfigMap, rev1 *corev1alpha1.ObjectSet) *corev1alpha1.ObjectSet {
+				objectSet := defaultObjectSetRev2(cm1, cm3, rev1)
+				objectSet.Spec.Phases[0].Class = "default"
+				objectSet.Spec.Phases[1].Class = "default"
+				return objectSet
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cm1 := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "cm-1",
+					Labels: map[string]string{"test.package-operator.run/test": "True"},
+				},
+			}
+			cmGVK, err := apiutil.GVKForObject(cm1, Scheme)
+			require.NoError(t, err)
+			cm1.SetGroupVersionKind(cmGVK)
+
+			cm2 := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cm-2",
+				},
+			}
+			cm2.SetGroupVersionKind(cmGVK)
+
+			objectSetRev1 := test.objectSetRev1(cm1, cm2)
+
+			cm3 := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cm-3",
+				},
+			}
+			cm3.SetGroupVersionKind(cmGVK)
+
+			objectSetRev2 := test.objectSetRev2(cm1, cm3, objectSetRev1)
+
+			ctx := logr.NewContext(context.Background(), testr.New(t))
+
+			require.NoError(t, Client.Create(ctx, objectSetRev1))
+			cleanupOnSuccess(ctx, t, objectSetRev1)
+
+			// Expect ObjectSet Rev1 to become Available.
+			require.NoError(t,
+				Waiter.WaitForCondition(ctx, objectSetRev1, corev1alpha1.ObjectSetAvailable, metav1.ConditionTrue))
+
+			// expect cm-1 to be present.
+			currentCM1 := &corev1.ConfigMap{}
+			require.NoError(t, Client.Get(ctx, client.ObjectKey{
+				Name: cm1.Name, Namespace: objectSetRev1.Namespace}, currentCM1))
+
+			// expect cm-2 to be present.
+			currentCM2 := &corev1.ConfigMap{}
+			require.NoError(t, Client.Get(ctx, client.ObjectKey{
+				Name: cm2.Name, Namespace: objectSetRev2.Namespace}, currentCM2))
+
+			// Create Revision 2
+			require.NoError(t, Client.Create(ctx, objectSetRev2))
+			cleanupOnSuccess(ctx, t, objectSetRev2)
+
+			// Expect ObjectSet Rev2 report not Available, due to failing probes.
+			require.NoError(t,
+				Waiter.WaitForCondition(ctx, objectSetRev2, corev1alpha1.ObjectSetAvailable, metav1.ConditionFalse))
+			availableCond := meta.FindStatusCondition(objectSetRev2.Status.Conditions, corev1alpha1.ObjectSetAvailable)
+			require.NotNil(t, availableCond, "Available condition is expected to be reported")
+			assert.Equal(t, "ProbeFailure", availableCond.Reason)
+
+			// expect cm-1 to still be present and now owned by Rev2.
+			require.NoError(t, Client.Get(ctx, client.ObjectKey{
+				Name: cm1.Name, Namespace: objectSetRev1.Namespace}, currentCM1))
+			currentCM1.GetOwnerReferences()
```

Comment:
```suggestion
			currentCM1.GetOwnerReferences()
assertIsController(t, objectSetRev2, currentCM1)
```

Reply:
Yeah, I was killing "assertIsController" because with phases, you cannot simply check if the object is owned by the ObjectSet anymore. It may also be owned by the phase object - which we don't even "see" from the test case.

I'll try to come up with a way to keep that check in.